src/dbms/handler.hpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <filesystem>
+#include <memory>
+#include <optional>
+#include <string_view>
+#include <unordered_map>
+
+#include "global.hpp"
+#include "utils/result.hpp"
+#include "utils/sync_ptr.hpp"
+
+namespace memgraph::dbms {
+
+/**
+ * @brief Generic multi-database content handler.
+ *
+ * @tparam TContext
+ * @tparam TConfig
+ */
+template <typename TContext, typename TConfig>
+class Handler {
+ public:
+  using NewResult = utils::BasicResult<NewError, std::shared_ptr<TContext>>;
+
+  /**
+   * @brief Empty Handler constructor.
+   *
+   */
+  Handler() {}
+
+  /**
+   * @brief Generate a new context and corresponding configuration.
+   *
+   * @tparam T1 Variadic template of context constructor arguments
+   * @tparam T2 Variadic template of config constructor arguments
+   * @param name Name associated with the new context/config pair
+   * @param args1 Arguments passed (as a tuple) to the context constructor
+   * @param args2 Arguments passed (as a tuple) to the config constructor
+   * @return NewResult
+   */
+  template <typename... T1, typename... T2>
+  NewResult New(std::string name, std::tuple<T1...> args1, std::tuple<T2...> args2) {
```

Comment:
Is there a reason we are deferring the construction of TContext and TConfig? 
Also https://en.cppreference.com/w/cpp/utility/make_from_tuple should make this code cleaner

Reply:
:D